src/components/EnterPinModal.tsx

```diff
@@ -0,0 +1,158 @@
+import React from 'react';
+import { Button, Modal, ModalHeader, ModalFooter, ModalBody, Row, Spinner } from 'reactstrap';
+import api from '../api'
+import { Device } from '../types';
+
+const digits = [
+  [7,8,9],
+  [4,5,6],
+  [1,2,3],
+]
+
+interface Props {
+	toggle(): any;
+  isOpen: boolean;
+  device: Device | null;
+}
+
+interface State {
+  pin: string;
+  submitting: boolean;
+  error: string | null;
+  pressed: number | null;
+}
+
+const initialState = {
+  pin: '',
+  submitting: false,
+  error: null,
+  pressed: null,
+}
+
+export default class EnterPinModal extends React.Component<Props, State> {
+  state: State = initialState;
+
+  async enterPin() {
+    const { pin, submitting } = this.state;
+    const { device } = this.props;
+    if (!submitting) {
+      try {
+        this.setState({ submitting: true })
+        await api.enterPin({ pin });
+        this.toggle()
+      } catch(error) {
+        this.setState({ 
+          error: error.message,
+          pin: '',
+          submitting: false,
+         });
+        // Prompt another pin so that user can try again  
+        // Annoying that typescript makes me check this ...
+        if (device) {
+          setTimeout(
+            () => api.promptPin({ path: device.path}),
+            1000
+          );
+        }
+      }
+    }
+  }
+
+  getStyles(digit: number) {
+    let style = {
+      height: '100px',
+      width: '100px',
+      backgroundColor: '#868e96',  // Bootstrap's "secondary" grey
+      margin: '10px',
+    }
+    if (this.state.pressed === digit) {
+      style['backgroundColor'] = '#ced4da'  // lighter grey
+    }
+    return style;
+  }
```

Comment:
This is a lot of code just for `:active` state, this would be a lot simpler with CSS (better yet LESS or SCSS) in my opinion. But if you want to keep the whole app CSS in JS-y, then that's fine. Might be nice to get a library for that though.